client/angular-dapp/src/app/login/login.component.html

```diff
@@ -19,30 +19,41 @@
         </button>
       </div>
     </div>
-    <ng-template #noDID>
-      <div *ngIf="unauthorized">
+    <ng-template #notAuthorized>
+      <ng-container *ngIf="roles.length > 0; else noRoles">
+        <p>Hello!</p>
         <p>
-          Unauthorized login response.
-          <br />
-          Please ensure that you have the necessary role claim.
+          You're not logged in because you do not have required roles. Please look below on what roles you're missing.
```

Comment:
I think if the roles are in the table, that means that the user has them but they are expired or revoked, right?
```suggestion
          You're not logged in because you do not have required roles.
          The table below may show roles which you have but are expired or revoked.
```

Reply:
yes, user enroled to them but they are expired/revoked/valid, depends how many roles were specified in ACCEPTED_ROLES variable.
Thanks, i will adjust text.